src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceContext.java

```diff
@@ -232,7 +252,18 @@ public GitHubSCMSourceContext withForkPRStrategies(Set<ChangeRequestCheckoutStra
         forkPRStrategies.addAll(strategies);
         return this;
     }
-
+    /**
+     * Defines the strategies used to notify Github of build status.
+     *
+     * @param strategies the strategies used to notify Github of build status.
+     * @return {@code this} for method chaining.
+     * @since TODO
+     */
+    @NonNull
+    public final GitHubSCMSourceContext withNotificationStrategies(Set<? extends AbstractGitHubNotificationStrategy> strategies) {
```

Comment:
Have this take a `List` and have the list replace the existing list.

Then have a `withNotificationStrategy` method that adds to the list if that *instance* is not already present (if you mandate that `AbstractGitHubNotificationStrategy` must implement `equals`, e.g. by declaring `equals` as `abstract` in `AbstractGitHubNotificationStrategy` then you could compare based on equality to make the list behave as a set (while still being a list).

The important part is that the notifications are ordered, a set will not give the ordering